crates/bevy_render/src/lib.rs

```diff
@@ -236,3 +242,48 @@ fn check_for_render_resource_context(context: Option<Res<Box<dyn RenderResourceC
         );
     }
 }
+
+fn window_icon_changed(
+    mut map: Local<HashMap<WindowId, Handle<Texture>>>,
+    textures: Res<Assets<Texture>>,
+    mut windows: ResMut<Windows>,
+    asset_server: Res<AssetServer>,
+) {
+    for window in windows.iter_mut() {
+        /* Insert new icon changed */
+        if let Some(WindowIcon::Path(path)) = window.icon() {
+            if let Some(handle) = map.get(&window.id()) {
+                if let Some(handle_path) = asset_server.get_handle_path(handle) {
+                    if handle_path.path() != path {
+                        map.insert(window.id(), asset_server.load(path.clone()));
+                    }
+                }
+            } else {
+                map.insert(window.id(), asset_server.load(path.clone()));
+            }
+        }
+
+        /* Poll load state of handle and set the icon */
+        if let Some(handle) = map.get(&window.id()) {
+            match asset_server.get_load_state(handle) {
+                LoadState::Loaded => {
+                    if let Some(texture) = textures.get(handle) {
+                        /* TODO: Why can I not just unwrap here? */
+                        let window_icon = WindowIcon::from(WindowIconBytes {
+                            bytes: texture.data.clone(),
+                            width: texture.size.width,
+                            height: texture.size.height,
+                        });
+                        window.set_icon(window_icon);
+
+                        map.remove_entry(&window.id());
```

Comment:
```suggestion
                        map.remove(&window.id());
```

`remove` should suffice here

Reply:
I also used the Entry API here.

Reply:
I still prefer `remove` unless you explicitly use the return value of `remove_entry` (which we don't in this case).